src/backend/distributed/planner/recursive_planning.c

```diff
@@ -337,6 +340,13 @@ RecursivelyPlanSubqueriesAndCTEs(Query *query, RecursivePlanningContext *context
 		RecursivelyPlanNonColocatedSubqueries(query, context);
 	}
 
+	/*
+	 * Logical planner cannot handle "local_table" [OUTER] JOIN "dist_table", or
+	 * a query with local table/citus local table and subquery. We convert local/citus local
+	 * tables to a subquery until they can be planned
+	 */
+	ConvertUnplannableTableJoinsToSubqueries(query, context);
```

Comment:
RecursivelyPlanLocalTableJoins sounds like it will recursively plan all local table joins, which might not be the case. I guess it is not too bad, but I would say lets think a bit more. We can go with that if we can't find a better alternative